src/msg_handler.c

```diff
@@ -0,0 +1,388 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <ctype.h>
+
+#include "msg_handler.h"
+#include "global_vars.h"
+#include "output.h"
+#include "led.h"
+#include "fan.h"
+#include "buzzer.h"
+
+static char* itoa(int value, char* string);
+
+int parse_msg(char* buf, char* response, int max_len)
+{
+    char temp_str[11] = {0};  // for doing itoa conversions
```

Comment:
`itoa(-2147483648)` will overflow this buffer! Need space for 11 chars plus nullterm.

Reply:
Yeah, I got the maths wrong. Have corrected.